### PR TITLE
Sync failures viewmodel

### PR DESCRIPTION
**Pull request by Odrakir**
Part of #1334 

This only adds the view model that exposes a simple list for the view controller in my next PR to display the items in a table view.

My doubts are in the way I await the observable from the interactor. By doing that I don't need to dispose anything?

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/ViewModels/SyncFailuresViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            this.interactorFactory = interactorFactory;
+        }
+
+        public override async Task Initialize()
+        {
+            await base.Initialize();
+
+            SyncFailures = await interactorFactory
+                .GetItemsThatFailedToSync().Execute()
+                .Select(i => i.ToList())
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/SyncFailuresViewModel.cs]: This is a bit weird - maybe you can change directly the interactor's return type to list.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/SyncFailuresViewModelTests.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+using Toggl.PrimeRadiant.Models;
+using Xunit;
+
+namespace Toggl.Foundation.Tests.MvvmCross.ViewModels
+{
+    public sealed class SyncFailuresViewModelTests
+    {
+        public abstract class SyncFailuresViewModelTest : BaseViewModelTests<SyncFailuresViewModel>
+        {
+
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/SyncFailuresViewModelTests.cs]: 🔥

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/SyncFailuresViewModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        }
+
+        public sealed class TheSyncFailuresProperty : SyncFailuresViewModelTest
+        {
+            [Fact, LogIfTooSlow]
+            public void ExecutesGetItemsThatFailedInteractor()
+            {
+                var mockedInteractor = Substitute.For<IInteractor<IObservable<IEnumerable<SyncFailureItem>>>>();
+                InteractorFactory.GetItemsThatFailedToSync().Returns(mockedInteractor);
+                ViewModel.Initialize();
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/SyncFailuresViewModelTests.cs]: Separate this with blank lines to achieve AAA (Arrange Act Assert)

File: Toggl.Foundation.MvvmCross/ViewModels/SyncFailuresViewModel.cs (hunk 1 above)
Comment (simonrozsival) [reply Toggl.Foundation.MvvmCross/ViewModels/SyncFailuresViewModel.cs]: Or maybe - do you need a list here? Isn't `IEnumerable` enough for the table?

Comment (Odrakir) [reply Toggl.Foundation.MvvmCross/ViewModels/SyncFailuresViewModel.cs]: I don't know, I'm guessing an `IEnumberable` is not enough as it doesn't have random access to its elements, right? I could do the conversion later though...

**Pull request merged**